VideoLocker/src/main/java/org/edx/mobile/view/CourseCombinedInfoFragment.java

```diff
@@ -345,23 +340,71 @@ private void populateFacePile(){
 
     }
 
-    private View generateAnnouncementView(AnnouncementsModel model){
+    private void populateAnnouncements(List<AnnouncementsModel> announcementsList) {
+        if(announcementsList !=null && announcementsList.size()>0) {
+            hideEmptyAnnouncementMessage();
 
-        View convertView = inflater.inflate(R.layout.row_announcement_list, null);
+            // stop existing views before we remove them from the layout
+            for (AnnouncementView view : announcementViewList) {
+                view.stop();
+            }
+            announcementViewList.clear();
+            announcementContainer.removeAllViews();
+
+            for (AnnouncementsModel m : announcementsList) {
+                AnnouncementView announcementView = generateAnnouncementView(m);
+                announcementViewList.add(announcementView);
+
+                announcementContainer.addView(announcementView.getView());
```

Comment:
it was a serious performance issue for both android and iphone as people tried to embed html style rich text into textview and none of android and iphone provide simple-to-use-api to achieve the goal. And some apps ends with a lot of embeded webview.    
and webview has some memory leak at that time.  I am not sure current status. 

but is that possible to combine all announcement into a single one? 
also,  should we use listview so weview can be reused?